### PR TITLE
chore(deploy): use given tab on <deployTab>

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Instead of ignoring the <tab> param and using the <activeTab> in the <deployTab> functionality, this ensures to actually using the given tab.

Follow up of Hour of Code, cf. https://github.com/bpmn-io/hour-of-code/blob/master/snippets/DeploymentTool_deployTab.md

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
I first thought about just renaming it to `deployActiveTab`, but if we want to use it somewhere in the future to deploy other tabs, I decided to refactor it this way.

File: client/src/plugins/deployment-tool/DeploymentTool.js [hunk 1]
```diff
@@ -41,17 +41,17 @@ export default class DeploymentTool extends PureComponent {
...
 
   componentDidMount() {
-    this.props.subscribe('app.activeTabChanged', activeTab => {
+    this.props.subscribe('app.activeTabChanged', ({ activeTab }) => {
       this.setState({ activeTab });
     });
   }
 
-  saveTab() {
+  async saveTab(tab) {
```
Comment (barmac) [inline client/src/plugins/deployment-tool/DeploymentTool.js]:
```suggestion
  saveTab(tab) {
```

We're returning a Promise anyway.

File: client/src/plugins/deployment-tool/DeploymentTool.js [hunk 2]
```diff
@@ -41,17 +41,17 @@ export default class DeploymentTool extends PureComponent {
...
     const {
       triggerAction
     } = this.props;
 
-    return triggerAction('save');
+    return await triggerAction('save-tab', { tab });
```
Comment (barmac) [inline client/src/plugins/deployment-tool/DeploymentTool.js]:
```suggestion
    return triggerAction('save-tab', { tab });
```

**Comment by barmac** [comment]
I'd rather call it a fix or even a feature if we allow to save not active tab.

**Comment by pinussilvestrus** [comment]
Added a small test case to save any tab via `triggerAction` and split it up to have an actual feature commit.

**Pull request merged**